lib/cartopy/mpl/geoaxes.py

```diff
@@ -1602,12 +1603,15 @@ def contour(self, *args, **kwargs):
         result = super().contour(*args, **kwargs)
 
         # We need to compute the dataLim correctly for contours.
-        bboxes = [col.get_datalim(self.transData)
-                  for col in result.collections
-                  if col.get_paths()]
-        if bboxes:
-            extent = mtransforms.Bbox.union(bboxes)
-            self.update_datalim(extent.get_points())
+        if _MPL_VERSION.release[:2] < (3, 8):
+            bboxes = [col.get_datalim(self.transData)
+                      for col in result.collections
+                      if col.get_paths()]
+            if bboxes:
+                extent = mtransforms.Bbox.union(bboxes)
+                self.update_datalim(extent.get_points())
+        else:
+            self.update_datalim(result.get_datalim(self.transData))
```

Comment:
Do we need to update the datalim here or is that already taken care of because it is a full collection that is added to the axes now? The other collection methods like scatter and pcolormesh don't update the datalims from their return objects. Maybe this has to do with the contour wrapping stuff and not getting the bounds until actually calling update_datalim()? This is mostly just my curiosity here.

Reply:
I saw several test failures when I tried not having it.  But now I look more closely I am questioning whether they matter.  For example

```
E           AssertionError: 
E           Arrays are not almost equal to 6 decimals
E           
E           Mismatched elements: 1 / 4 (25%)
E           Max absolute difference: 1.
E           Max relative difference: 0.00555556
E            x: array([-179.,  180.,  -25.,   25.])
E            y: array([-180,  180,  -25,   25])
```
which comes from the second assertion here.  So it fails because the non-transformed-first version is now the same as the transformed-first version.  Without much context, more consistency seems like an improvement?

https://github.com/SciTools/cartopy/blob/41880a82d8a6e9e9aa57515070e29e06bfddc7ac/lib/cartopy/tests/mpl/test_contour.py#L137-L146

There are also some image test failures which, by eye, look like they might be because of a similar extent change to the above.

We would need to add in something like `self.ignore_existing_data_limits = False` to cover my special dodgy latitude case from #2129.

Reply:
Ahh, OK. I think we definitely need this then, thanks for checking.

The two cases should definitely be different and that is preferable in these cases because their extents truly are different. The gallery kind of shows that if we force the extent to be global on the transformed-first one there is a gap and the extent is only -179/179 and doesn't touch the extreme edges. But, if we let the path logic do its magic it should be out to -180 and fill that gap.

https://scitools.org.uk/cartopy/docs/latest/gallery/scalar_data/contour_transforms.html#sphx-glr-gallery-scalar-data-contour-transforms-py